src/Xenial.Tasty.Transports.NamedPipes/Transports/NamedPipesTranport.cs

```diff
@@ -0,0 +1,37 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.IO.Pipes;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+using Xenial.Delicious.Remote;
+
+namespace Xenial.Delicious.Transports
+{
+    public static class NamedPipesTranport
+    {
+        public static Task<TransportStreamFactory?> CreateNamedPipeTransportStream(CancellationToken token = default)
+        {
+            var connectionType = Environment.GetEnvironmentVariable(EnvironmentVariables.InteractiveConnectionType);
+            if (string.Equals(connectionType, "NamedPipes", StringComparison.InvariantCultureIgnoreCase))
```

Comment:
If the environment variable is not found or is doesn't equal to NamedPipes... should it throw? How does the user knows it failed?

Reply:
This is a fall through protocol, so the next transport registered will look if a connection can be established. (For example Websockets).
Not every session is interactive (this is what I'm currently working on) so this needs to be handled by the `ConnectToRemoteStreamMiddleware` and throw there.
I will convert this to a `ConnectionString` property.

As for now the user never will get involved into building this connection string though, only during an internal abstraction.